### DISCUSSION TITLE
Welcome to fpdf2 Discussions!

### DISCUSSION THREAD

**Discussion by Lucas-C** [py-pdf/fpdf2#233] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Lucas-C** [comment]
Hi!

I'm opening this discussions section as an experiment 😉

I'm a French software engineer, based around Angers. I'm the maintainer of fpdf2 since last winter.
I also make games, sometimes [in PDF format](https://lucas-c.itch.io/undying-dusk).

If you have questions or ideas to improve fpdf2, feel free to share theme here!

**Reply by sir-george2500** [reply]
Hello I will like to contribute to this project is there any get started page I can look at

**Reply by Lucas-C** [reply]
Hi @sir-george2500 🙂

Welcome to the community of `fpdf2` contributors, and thanks for volunteering!

We do not have a proper "Getting started" page, but those pages are a good starting point:
* https://py-pdf.github.io/fpdf2/Development.html
* https://py-pdf.github.io/fpdf2/Internals.html

We have some issues tagged `good first issues` aimed at newcomers:
https://github.com/py-pdf/fpdf2/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

This one could be a good pick: https://github.com/py-pdf/fpdf2/issues/1272

**Comment by Gui85150** [comment]
Hello, 
I use python 3.9 and fpdf 2.6

I want use write_html with html= " 20 &euro;"
The document *.html is well done and I see in it the symbol €
But when i try pdf.write_html(html) i have an error
because the character € isn't in the font.
How Can i do for conserving my code html and see the € in the pdf

Thanks youn and sorry for my Bad english

**Reply by Lucas-C** [reply]
Hi @Gui85150!

You will simply have to use custom font 😊

There is a example of using a custom TTF font here in our documentation:
https://pyfpdf.github.io/fpdf2/Unicode.html#example

`.ttf` font files are very common, and most of them should render the € symbol: https://fr.wikipedia.org/wiki/TrueType

**Reply by Gui85150** [reply]
hi Lucas,
Thanks for your response.

I've installed ArialUni and this font is correctly installed because if i try to put in my script the line simple html:

               pdf.write_html("""<p>20 &euro;</p>""")

ther's no matter and the pdf is well done So my font is correctly installed.
But if i try next with the other html with contain table with head, i have an error with those lines

                 Traceback (most recent call last):
                    pdf.write_html(html)
                 File "XXX\lib\site-packages\fpdf\fpdf.py", line 417, in write_html
                    h2p.feed(text)
                  File "XXX\lib\site-packages\fpdf\html.py", line 770, in feed
                    super().feed(data)
                  File "XXX\lib\html\parser.py", line 110, in feed
                    self.goahead(0)
                  File "XXX\lib\html\parser.py", line 162, in goahead
                    self.handle_data(unescape(rawdata[i:j]))
                  File "XXX\lib\site-packages\fpdf\html.py", line 280, in handle_data
                    self._insert_td(data)
                  File "XXX\lib\site-packages\fpdf\html.py", line 344, in _insert_td
                    self.set_style("B", True)
                  File "XXX\lib\site-packages\fpdf\html.py", line 795, in set_style
                    self.pdf.set_font(style=style)
                  File "XXX\lib\site-packages\fpdf\fpdf.py", line 1914, in set_font
                    raise FPDFException(
                fpdf.errors.FPDFException: Undefined font: arialuniB - Use built-in fonts or FPDF.add_font() beforehand

So i try with a document html without bold ( table without head) and the document is well done but with a message in the console :

                 Font MPDFAA+ArialUnicodeMS is missing the following glyphs:

So there a matter with my font and using bold in the html document ....

**Reply by Lucas-C** [reply]
Hi @Gui85150 

I'm sorry but I'm not quite sure on what the source of your problem could be...
Have you called `add_font()` with the path to `ArialUni.ttf` provided as `fname`?

Could you please [open a bug report](https://github.com/PyFPDF/fpdf2/issues/new?labels=bug,html&template=bug_report.md) and include some **minimal Python code** reproducing your issue:
```python
from fpdf import FPDF

pdf = FPDF()
...
```
If you don't know how to build a minimal reproducible example, please check this tutorial: https://stackoverflow.com/help/minimal-reproducible-example
